### ISSUE TITLE
A periodic task which can be scheduled manually

### ISSUE THREAD

**Issue by mindojo-victor** [coleifer/huey#386]
Continuation of #277 

> I would:
> 
> ```python
> def one_func():
>     pass
> 
> one_func_task = huey.task()(one_func)
> one_func_cron = huey.periodic_task(crontab(...))(one_func)
> ```

It's not working because of
https://github.com/coleifer/huey/blob/533c681519f10b4e354e567b6d8790f07f8d8a4d/huey/registry.py#L26-L31

The call `huey.periodic_task` doesn't add the function to `_periodic_tasks` because it is already present in the registry as a task on demand.

**Comment by coleifer** [comment]
Have you tried specifying a `name` for the task?

```python
one_func_task = huey.task()(one_func)
one_func_cron = huey.periodic_task(crontab(...), name='one_func_cron')(one_func)
```

**Comment by mindojo-victor** [comment]
This should work. Thanks for an workaround!

However, I think it's not uncommon for people to want the same task be called on demand and periodically. And using your solution is coupling an implementation detail.
I think if this is not supported, a user friendly approach would be to raise an exception or show a warning when a periodic task failed to be added to the list of periodic tasks, otherwise developers could find out that it doesn't work in production, like it was in my case.
Or a better approach (IMO) would be to fix this side effect.
Or provide a method to explicitly add/remove periodic tasks to/from `TaskRegistry._periodic_tasks`, because passing `validate_datetime` to `huey.periodic_task` essentially does the same, but with hidden and uncontrollable logic.

**Comment by coleifer** [comment]
This is now fixed: 78856b07936efc80d56b1b1ac147f268de46d6ed

An error will now be raised if you attempt to register (or decorate) a task that conflicts with the name of a previously-registered task. So no more silent failures.

A side-effect of how I implemented this is that the task serialization scheme has changed to accommodate the new naming scheme. So these changes should be considered backwards-incompatible in terms of the serialization format used.
